core/src/ProblemManager.cc

```diff
@@ -233,22 +269,36 @@ void ProblemManager::calculateNodalMomentum(
     int space_dim = d_mesh->spatialDimension();
     int nodes_per_cell = d_mesh->nodesPerCell();
     int node_id = 0;
+    int num_nodes = d_mesh->totalNumNodes();
+
+    std::vector<std::vector<std::array<double,3> > > node_p_local(
+        d_thread_count, std::vector<std::array<double,3> > (
+	num_nodes, std::array<double,3>() ) );
 
-    // Reset the momentum
+    // Initialize local storage to zero.
+    for ( auto& thread : node_p_local )
+	for ( auto& node_mom : thread )
+            std::fill( node_mom.begin(), node_mom.end(), 0.0 );
```

Comment:
actually does this work? would you not have to fill with `{0.0,0.0,0.0}` or something similar?

Reply:
Note that this was fine for the loop below because `std::fill` is called on a vector of scalars whereas here we are calling it on a vector of arrays

Reply:
I think that the double for loop will accomodate that because the inner loop that is filling is a vector of scalars